src/caffe/test/test_gradient_based_solver.cpp

```diff
@@ -565,7 +581,9 @@ class SGDSolverTest : public GradientBasedSolverTest<TypeParam> {
 
  protected:
   virtual void InitSolver(const SolverParameter& param) {
-    this->solver_.reset(new SGDSolver<Dtype>(param));
+    SolverParameter new_param = param;
+    new_param.set_type("SGD");
```

Comment:
This should no longer be needed as of #5009. Please drop these `set_type0` to check.